ge/org.osate.ge/src/org/osate/ge/internal/ui/editor/actions/SelectAllAction.java

```diff
@@ -0,0 +1,42 @@
+package org.osate.ge.internal.ui.editor.actions;
+
+import java.util.Objects;
+
+import org.eclipse.gef.ui.actions.WorkbenchPartAction;
+import org.eclipse.graphiti.mm.pictograms.PictogramElement;
+import org.eclipse.ui.actions.ActionFactory;
+import org.osate.ge.internal.diagram.runtime.DiagramElement;
+import org.osate.ge.internal.graphiti.diagram.GraphitiAgeDiagram;
+import org.osate.ge.internal.ui.editor.AgeDiagramEditor;
+
+import com.google.common.base.Predicates;
+
+/**
+ * Action that selects all diagram elements.
+ *
+ */
+public class SelectAllAction extends WorkbenchPartAction {
+	private final AgeDiagramEditor editor;
+
+	public SelectAllAction(final AgeDiagramEditor editor) {
+		super(editor);
+		this.editor = Objects.requireNonNull(editor);
+		setId(ActionFactory.SELECT_ALL.getId());
+	}
+
+	@Override
+	public void run() {
+		final GraphitiAgeDiagram gad = editor.getGraphitiAgeDiagram();
+		final PictogramElement[] connectionPictogramElements = editor.getAgeDiagram().getAllDiagramNodes()
+				.filter(dn -> dn instanceof DiagramElement).map(DiagramElement.class::cast)
```

Comment:
dn -> dn instanceof DiagramElement could be DiagramElement.class::isInstance (more consistent with DiagramElement.class::cast).  But it is up to you if you want to change it.